Added _calcChainableHelper for mvOverlapExpression and mvContainsExpression

mujinss opened the pull request:
Jira Task: [IMPLY-29907](https://implydata.atlassian.net/browse/IMPLY-29907)

Context:
The error described in the ticket occurs because clicking on the color legend triggers `FilterClause.matches()` , which calls Expression.calc() inside. (https://github.com/implydata/imply-ui/blob/master/packages/pivot-common/src/models/filter-clause/filter-clause.ts#L469). For `MvOverlapExpression` and `MvContainsExpression` in Plywood, this function is not implemented.

[IMPLY-29907]: https://implydata.atlassian.net/browse/IMPLY-29907?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

jgoz commented:
Is there a way to expand the test cases to cover this functionality in `test/simulate/simulateDruid*` tests?

mujinss commented:
> Is there a way to expand the test cases to cover this functionality in `test/simulate/simulateDruid*` tests?

I have added two tests for expressions, but not quite sure how to incorporate those functions into simulate tests. Those simulates seem to be testing on query generation, but I haven't found out an example how `calc` functions are used in query generation.
A more important question.... I only know that those are the methods missing in the two classes, but only sure if the way I implemented them are the correct content to fill in there 😅

lorem--ipsum commented:
This looks good to me, but ideally @vogievetsky would validate it too

vogievetsky commented on src/expressions/mvContainsExpression.ts:
```diff
@@ -58,6 +59,16 @@ export class MvContainsExpression extends ChainableExpression {
+      typeof operandValue === 'string'
+        ? [operandValue]
+        : Array.isArray(operandValue)
+        ? [...operandValue]
```
no need to copy the array here, right?

vogievetsky commented on src/expressions/mvOverlapExpression.ts:
```diff
@@ -58,6 +59,16 @@ export class MvOverlapExpression extends ChainableExpression {
+      typeof operandValue === 'string'
+        ? [operandValue]
+        : Array.isArray(operandValue)
+        ? [...operandValue]
```
ditto
